amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/LifecycleConfig.java

```diff
@@ -46,4 +46,10 @@
      */
     private AggregatorUtil aggregatorUtil = new AggregatorUtil();
 
+    /**
+     * TaskExecutionListener to be used to handle events during task execution lifecycle for a shard.
+     *
+     * <p>Default value: {@link NoOpTaskExecutionListener}</p>
+     */
+    private TaskExecutionListener taskExecutionListener = new NoOpTaskExecutionListener();
```

Comment:
Is the listener the same for all ShardConsumers? It would be better to have a factory and create a listener per ShardConsumer.

Reply:
I think having a single instance of the listener is fine.  The implementations of listener will need to handle that their singleton state.